### PR TITLE
Separate Parsec ItemNotFoundError into two

### LINKED ISSUE #4187: cylc config --sparse: item not found
`cylc config --sparse -i ITEM` aborts with an error if a legal `ITEM` is not explicitly-set: 

```console
$ cylc config -i 'task mail'           
from = 
to = 

$ cylc config -i 'task mail' --sparse 
ItemNotFoundError: item not found: task mail
```

**Question:** should this return:
- error status - `ITEM` not found in sparse config
- success status but no result - `ITEM` legal but not set

IMO, the latter is more sensible.

### PR DESCRIPTION
**Pull request by wxtim**
**Before**
ItemNotFoundError was returned if
- The item wasn't defined in the sparse config.
- The item wasn't available to be defined, because the item was mis-spelled or totally non-existant.

**After**
ItemNotFoundError - Where an item could be defined, but isn't (in sparse mode).
NotAConfigItemError - Where an item is not part of a config to be defined.

These changes close #4187 

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [ ] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.py` and
  `conda-environment.yml`.
- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] No documentation update required.

### CODE REVIEW

**Comment by MetRonnie** [comment]
Something very odd has happened to this branch (discontinuity between the commit named `fixed the logic further` and `pr/wxtim/4471-1`): 

image

I'm going to try re-merging master into it and force-pushing

**Comment by MetRonnie** [comment]
I've also thought of a way to simplify the logic - opened a PR against this one at https://github.com/wxtim/cylc/pull/32

**Comment by MetRonnie** [comment]
Rebased wxtim#32 on latest changes

**Comment by MetRonnie** [comment]
(Still waiting for outcome of wxtim#32)

**Comment by wxtim** [comment]
> (Still waiting for outcome of [wxtim#32](https://github.com/wxtim/cylc/pull/32))

Sorry, missed that you'd not made the changes directly to the branch but had created a PR to PR.

File: CHANGES.md [hunk 1]
```diff
@@ -104,9 +104,14 @@ now accepts ``&`` and ``|`` as valid line breaks in the same way as ``=>``.
...
 `CYLC_WORKFLOW_ID` shorn of any trailing `runX`.
 
+[#4471](https://github.com/cylc/cylc-flow/pull/4471) - Users now get a different
+error for a config item that isn't valid, to one that isn't set.
+
 [#4457](https://github.com/cylc/cylc-flow/pull/4457) - Cylc 8
 `cycle point time zone` now defaults to UTC, except in Cylc 7 compatibility mode.
 
+[#4457](https://github.com/cylc/cylc-flow/pull/4457) - Cylc 8
+`cycle point time zone` now defaults to UTC, except in Cylc 7 compatibility mode.
```
Comment (MetRonnie) [inline CHANGES.md:114]: Duplicate entry
```suggestion
```

**Review by MetRonnie — APPROVED** [review]
Happy assuming tests pass

**Pull request merged**